src/core/modules/SolarSystem.cpp

```diff
@@ -1543,88 +1544,317 @@ void SolarSystem::computePositions(StelCore *core, double dateJDE, PlanetP obser
+			// Move to external threads, but also run a part in the main thread. The index 'availableThreads' is just the last group of objects.
+			for (int stride=0; stride<extraThreads; stride++)
+			{
+				auto future=QtConcurrent::run(plCompLoopZero, stride);
+				futures.append(future);
+			}
+			plCompLoopZero(extraThreads);
+
+			// Now the list is being computed by other threads. we can just wait sequentially for completion.
+			for(auto f: futures)
+				f.waitForFinished();
+			futures.clear();
+
+			const Vec3d &obsPosJDE=observerPlanet->getHeliocentricEclipticPos();
+
+			// 2.&3.: For higher accuracy, we now make two iterations of light time and aberration correction. In the final
 			// round, we also compute rotation data.  May fix sub-arcsecond inaccuracies, and optionally apply
 			// aberration in the way described in Explanatory Supplement (2013), 7.55.  For reasons unknown (See
 			// discussion in GH:#1626) we do not add anything for the Moon when observed from Earth!  Presumably the
 			// used ephemerides already provide aberration-corrected positions for the Moon?
-			Vec3d planetPos = p->getHeliocentricEclipticPos();
-			double lightTimeDays = (planetPos-obsPosJDE).norm() * (AU / (SPEED_OF_LIGHT * 86400.));
-			const bool needToApplyAberration = withAberration && (!observerIsEarth || p != getMoon());
-			Vec3d aberrationPush(0.);
-			if(needToApplyAberration)
-				aberrationPush=lightTimeDays*aberrationPushSpeed;
-			p->computePosition(obs, dateJDE-lightTimeDays, aberrationPush);
+			const Vec3d aberrationPushSpeed=observerPlanet->getHeliocentricEclipticVelocity() * core->getAberrationFactor();
 
-			// Extra accuracy with another round. Not sure if useful. Maybe hide behind a new property flag?
-			planetPos = p->getHeliocentricEclipticPos();
-			lightTimeDays = (planetPos-obsPosJDE).norm() * (AU / (SPEED_OF_LIGHT * 86400.));
-			if(needToApplyAberration)
-				aberrationPush=lightTimeDays*aberrationPushSpeed;
-			// The next call may already do nothing if the time difference to the previous round is not large enough.
-			p->computePosition(obs, dateJDE-lightTimeDays, aberrationPush);
-
-			const auto update = &RotationElements::updatePlanetCorrections;
-			if      (p->englishName==L1S("Moon"))    update(dateJDE-lightTimeDays, RotationElements::EarthMoon);
-			else if (p->englishName==L1S("Mars"))    update(dateJDE-lightTimeDays, RotationElements::Mars);
-			else if (p->englishName==L1S("Jupiter")) update(dateJDE-lightTimeDays, RotationElements::Jupiter);
-			else if (p->englishName==L1S("Saturn"))  update(dateJDE-lightTimeDays, RotationElements::Saturn);
-			else if (p->englishName==L1S("Uranus"))  update(dateJDE-lightTimeDays, RotationElements::Uranus);
-			else if (p->englishName==L1S("Neptune")) update(dateJDE-lightTimeDays, RotationElements::Neptune);
-
-			if(p != observerPlanet)
+			std::function<void (int)> plCompLoopOne = [=](int offset){
```

Comment:
No need in `std::function`, just keep it a lambda (`auto`). It will be converted into a Qt's callable anyway, so `std::function` is just an unnecessary overhead.

Reply:
According to QtCreator's tooltip, auto resolves to this type. I don't like auto, the written type is IMO clearer to read. Or does auto resolve differently on Linux?

Reply:
A lambda is never an `std::function`. You can see it if you compile and query its type in a variable viewer or simply check `sizeof` it compared to the case when you've assigned it to an `std::function`. Each lambda has a unique type, while `std::function` is a type-deleting wrapper over any kind of callable: a lambda, a named class with `operator()`, a function pointer, etc.

> the written type is IMO clearer to read

It might be when you know the exact type (and for lambda there's no name for the type, except `decltype(myLambda)` which already needs you to have used `auto` in order to use `myLambda` further). And this is the pitfall where you can get inefficiencies like e.g. doing a copy instead of getting a reference.

Reply:
OK, thanks for the explanation. I change these functions to auto.